src/seedu/addressbook/commands/Command.java

```diff
@@ -70,4 +70,14 @@ public int getTargetIndex() {
     public void setTargetIndex(int targetIndex) {
         this.targetIndex = targetIndex;
     }
+
+    public boolean isMutating(){
+
+        if (this.getClass().equals(AddCommand.class) || this.getClass().equals(ClearCommand.class)) {
+            return true;
```

Comment:
as discussed during the tutorial, it is not recommended to include default behavior.
> (this.getClass().equals(AddCommand.class) || this.getClass().equals(ClearCommand.class))

Also it is not good to do something like this as it defeats the purpose of polymorphism